test/e2e/pods.go

```diff
@@ -35,19 +35,10 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-// createNamespaceIfDoesNotExist ensures that the namespace with specified name exists, or returns an error
-func createNamespaceIfDoesNotExist(c *client.Client, name string) (*api.Namespace, error) {
-	namespace, err := c.Namespaces().Get(name)
-	if err != nil {
-		namespace, err = c.Namespaces().Create(&api.Namespace{ObjectMeta: api.ObjectMeta{Name: name}})
-	}
-	return namespace, err
-}
-
 func runLivenessTest(c *client.Client, podDescr *api.Pod, expectRestart bool) {
-	ns := "e2e-test-" + string(util.NewUUID())
-	_, err := createNamespaceIfDoesNotExist(c, ns)
-	expectNoError(err, fmt.Sprintf("creating namespace %s", ns))
+	namespace, err := createTestingNS("pods-liveness", c)
```

Comment:
Here is a reasonsable canonical example:

https://github.com/GoogleCloudPlatform/kubernetes/blob/85ade1db0a1918f251909912c1c877b78be17c3e/test/e2e/service_latency.go#L40

cc: @lavalamp in case he has any further suggestions.